docs/specification.md

```diff
@@ -0,0 +1,1280 @@
+  and sets its minting allowance to 0. Once minter is removed it will no
+  longer be able to mint or burn tokens.
+
+- Sender must be master minter.
+
+- Minter address must present in sender's minter list.
+
+**setMintingAllowance**
+
+Types
+```
+tokenId = nat
+
+setMintingAllowanceParam =
+  ( address :sender
+  , address :minter
+  , tokenId :tokenId
+  , nat     :amount
+  )
+
+setMintingAllowance = setMintingAllowanceParam
+```
+
+Parameter (in Michelson)
+```
+(pair %setMintingAllowance
+  (address %sender)
+  (pair
+    (address %minter)
+    (pair
+      (nat %tokenId)
+      (nat %amount)
+)))
+```
+
+- Set the amount of allowed minting allowance for an address with some
+  specific token type.
+
+- Minter must present in sender's minter list.
+
+- Sender and minter must be whitelisted.
+
+**getMintingAllowance**
+
+Types
+```
+tokenId = nat
+
+getMintingAllowanceRequest =
+  ( address :owner
+  , tokenId :tokenId
+  )
+
+getMintingAllowanceResponse =
+  ( getMintingAllowanceRequest :request
+  , nat                        :allowance
+  )
+
+getMintingAllowanceParam =
+  ( list getMintingAllowanceRequest             :requests
+  , contract (list getMintingAllowanceResponse) :callback
+  )
+
+getMintingAllowance = getMintingAllowanceParam
+```
+
+Parameter (in Michelson):
+```
+(pair %getMintingAllowance
+  (list %requests
+    (pair
+      (address %owner)
+      (nat %tokenId)
+    )
+  )
+  (contract %callback
+    (list
+      (pair
+        (pair %request
+          (address %owner)
+          (nat %tokenId)
+        )
+        (nat %allowance)
+      )
+    )
+  )
+)
+```
+
+- Returns current minting allowance for address with a specific token type.
+
+**getTotalMinted**
+
+Types
+```
+tokenId = nat
+
+getTotalMintedResponse =
+  ( tokenId :tokenId
+  , nat     :totalMinted
+  )
+
+getTotalMintedParam =
+  ( list tokenId                           :tokenIds
+  , contract (list getTotalMintedResponse) :callback
+  )
+
+getTotalMinted = getTotalMintedParam
+```
+
+Parameter (in Michelson)
+```
+(pair %getTotalMinted
+  (list %tokenIds nat)
+  (contract %callback
+    (list
+      (pair %getTotalMintedResponse
+        (nat %tokenId)
+        (nat %totalMinted)
+      )
+    )
+  )
+)
+```
+- Returns current amount of minted coins for an address with a specific token
+  type.
+
+-- TODO: clarify under which conditions and how should
+-- token reset this allowance
+
+**setMintingAllowanceResetInterval** timestamp
+
+Types
+```
+setMintingAllowanceResetInterval =
+  ( tokenId   :tokenId
+  , timestamp :interval
+  )
+
+```
+
+Parameter (in Michelson)
+```
+(pair %setMintingAllowanceResetInterval
+  (address %sender)
+  (timestamp %interval)
+)
+```
+
+- Set the default interval of minting allowance reset.
+
+- Set to 0 to make disable minting allowance reset by stablecoin token.
+
+**mint (TBD)**
+
+Types
+```
+tokenId = nat
+
+mintParam
+  ( address :sender
+  , address :minter
```

Comment:
You can use `SENDER` to get the transaction sender and check whether it is a `minter`.
I don't get why you use `minter` here at all because AFAIU minter is the sender in all cases.

But I don't get why you do not include the token recipient in the transaction param. Who will receive the tokens in this case?

Reply:
Yep, it should be renamed to `:masterMinter` and `:recipient`.